### DISCUSSION TITLE
General feedback about the card

### DISCUSSION THREAD

**Discussion by ovizii** [thlucas1/spotifyplus_card#11] [General]
The “progress-bar” needs some outline. Compare it with the volume bar, I can clearly see how long the volume bar is and click to select the volume. With the bar at the top I pretty much have to randomly click, hoping I hit the right spot. Maybe give us the option to select a color for the bar starting from 

![image](https://github.com/user-attachments/assets/ed57291f-9015-4377-b607-9093f97487c6)

Also, there is something else I am interested in:

When I check the settings of my Spotify app, I have enabled the “autoplay” feature so whenever I start playing a track, it will keep continuing with similar tracks. Or at least it seemed that way. I might be wrong, I'm just a very casual Spotify User.
Could you add such an option to the card? Currently, if I start playing a track from the card, it will simply stop after that track.

**Comment by ovizii** [comment]
On the Spotify app, I see info like: playing from playlist or playing from album, etc. 
Is there a way to see that info on the card?
Is there a way to see the queue on the card? Sometimes I like to peek ahead :-)

**Reply by thlucas1** [reply]
I added an outline to the progress bar - good idea, thanks for that!

There is no queue support as of yet - future enhancement possibility on that.

As for the playlist name, album name, etc, you can use the [Title Formatter Options](https://github.com/thlucas1/spotifyplus_card/wiki/Configuration-Options#title-formatter-options) to display just about anything that you want in the header area of the Player.  You just have to modify the Player configuration options ...

To do that:
1) edit the card configuration.  I would suggest you edit the card while playing a track (or playlist) so that you can see the title formatter values as you enter them.
2) select the `Player` tab to display Player section settings.
3) Adjust the following Player Header Status area settings, using the title formatter options.
  - Title text
  - Artist and Track Info
  - Album Info

For example, you could set the "Title text" field to:
Current Playlist: {player.sp_playlist_name} and it would render as "Current Playlist: My Playlist Name"

Note that the "Artist and Track Info" setting does not have to contain an artist or track - it can be any of the title formatter options.  Same for the Title Text and Album Info.  I probably should have labelled these Title text, Sub-Title1 Text and Sub-Title2 Text to make that more clear.

### Title Formatter Options
- {player.name} media player name (e.g. "LivingRoom Soundbar").
- {player.friendly_name} media player friendly name (e.g. "LivingRoom Soundbar").
- {player.media_album_name} currently playing media content album name (e.g. "It Is Well").
- {player.media_artist} currently playing media content artist (e.g. "Kutless").
- {player.media_title} currently playing media content title (e.g. "What Faith Can Do").
- {player.media_track} currently playing media content track (e.g. "What Faith Can Do").
- {player.state} media player current state (e.g. "playing", "off", etc).
- {player.sp_context_uri} Spotify uri for the currently playing context (e.g. "spotify:album:5ovMagVMhha9qq0OYDzXFm").
- {player.sp_playlist_name} playlist name being played, if the current context is a playlist (e.g. "Daily Mix 1").
- {player.sp_playlist_name_title} playlist name being played along with a "Playlist: " prefix, if the current context is a playlist (e.g. "Playlist: -Daily Mix 1").
- {player.sp_playlist_uri} playlist uri being played, if the current context is a playlist (e.g. "spotify:playlist:37i9dQZF1EYkqdzj48dyYq").

**Comment by thlucas1** [comment]
Forgot to mention, I will have to look into the “autoplay” feature.  I thought that was automatically done by the Spotify API, but it looks like it's not.  Future enhancement if so.

**Comment by thlucas1** [comment]
@ovizii 
Would you mind submitting [Feature Requests](https://github.com/thlucas1/spotifyplus_card/issues/new?assignees=&labels=enhancement&projects=&template=feature_request.yml) for the following?  That way I know who to notify if / when the feature is implemented:

- When I check the settings of my Spotify app, I have enabled the “autoplay” feature so whenever I start playing a track, it will keep continuing with similar tracks. Or at least it seemed that way. I might be wrong, I'm just a very casual Spotify User.
Could you add such an option to the card? Currently, if I start playing a track from the card, it will simply stop after that track.

- Is there a way to see the Spotify play queue on the card? Sometimes I like to peek ahead :-)

Loving the feedback; keep it coming, and thank you for the help.

**Reply by ovizii** [reply]
https://github.com/thlucas1/spotifyplus_card/issues/12 
https://github.com/thlucas1/spotifyplus_card/issues/13

**Reply by thlucas1** [reply]
Thank you!
